components/accelerator/nvidia/peermem/component.go

```diff
@@ -5,42 +5,60 @@ package peermem
 import (
 	"context"
 	"fmt"
-	"strconv"
 	"time"
 
 	"github.com/leptonai/gpud/components"
 	nvidia_peermem_id "github.com/leptonai/gpud/components/accelerator/nvidia/peermem/id"
-	"github.com/leptonai/gpud/components/dmesg"
-	"github.com/leptonai/gpud/pkg/common"
 	nvidia_common "github.com/leptonai/gpud/pkg/config/common"
+	events_db "github.com/leptonai/gpud/pkg/events-db"
 	"github.com/leptonai/gpud/pkg/log"
 	nvidia_query "github.com/leptonai/gpud/pkg/nvidia-query"
 	"github.com/leptonai/gpud/pkg/query"
 )
 
 func New(ctx context.Context, cfg nvidia_common.Config) (components.Component, error) {
+	eventsStore, err := events_db.NewStore(
+		cfg.Query.State.DBRW,
+		cfg.Query.State.DBRO,
+		events_db.CreateDefaultTableName(nvidia_peermem_id.Name),
+		3*24*time.Hour,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	cctx, ccancel := context.WithCancel(ctx)
+	w, err := newWatcher(cctx, eventsStore)
+	if err != nil {
+		ccancel()
+		return nil, err
+	}
+
 	if nvidia_query.GetDefaultPoller() == nil {
+		ccancel()
 		return nil, nvidia_query.ErrDefaultPollerNotSet
 	}
 
 	cfg.Query.SetDefaultsIfNotSet()
-
-	cctx, ccancel := context.WithCancel(ctx)
 	nvidia_query.GetDefaultPoller().Start(cctx, cfg.Query, nvidia_peermem_id.Name)
```

Comment:
I think we could remove it?

Reply:
commented... still need it for lsmod polling. We will deprecate this in the following PRs